include/rcutils/types/hash_map.h

```diff
@@ -156,7 +156,7 @@ rcutils_get_zero_initialized_hash_map();
  * ```
  *
  * \param[inout] hash_map rcutils_hash_map_t to be initialized
- * \param[in] initial_capacity the amount of initial capacity for the hash_map
+ * \param[in] initial_capacity the amount of initial capacity for the hash_map - this must be greater than zero and a power of 2
```

Comment:
I think this is a silly thing to do, now everyone that calls it has to do this calculation (it isn't trivial btw). Instead this function should take this as a request and document that it's rounding up to the nearest power of 2, not require the user to do this...

See what @methylDragon is having to do in his new pr:

https://github.com/ros2/rosidl/pull/728/files#diff-f7fe239aa4ebdf12a882356e1ef293b766aced118d5a9b59e487efb08f08c41aR45-R57

btw, the example, a few lines above uses 10, which isn't a power of two 🤦‍♂️

Reply:
> I think this is a silly thing to do, now everyone that calls it has to do this calculation (it isn't trivial btw). Instead this function should take this as a request and document that it's rounding up to the nearest power of 2, not require the user to do this...

That's fair.  There were only two users of this before, and they were both trivial to fix, so I just did it this way.  But I'd be fine modifying this so it does the calculation internally.

> btw, the example, a few lines above uses 10, which isn't a power of two man_facepalming

I just missed that example in the documentation.  All of the code does it properly.  I can fix that as well.

Reply:
See https://github.com/ros2/rcutils/pull/420